jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java

```diff
@@ -62,19 +62,26 @@
 abstract class AbstractFeignJaxRsClientBuilder {
 
     private final ClientConfiguration config;
+    /**
+     * The primary URI used to bootstrap the Feign client; this is the URI used by Feign to create an OkHttp call. Note
+     * that the {@link MultiServerRetryInterceptor} replaces this URI with a random URI from the client configuration
+     * before making the request.
+     */
+    private final String primaryUri;
 
     AbstractFeignJaxRsClientBuilder(ClientConfiguration config) {
-        this.config = config;
         Preconditions.checkArgument(config.maxNumRetries() == 0,
                 "Connection-level retries are not supported by %s", JaxRsClient.class.getSimpleName());
+        Preconditions.checkArgument(!config.uris().isEmpty(), "Must provide at least one service URI");
```

Comment:
i think we'll have to do the filtering for valid/invalid configurations on the witchcraft side. this preconditions here only makes an existing hidden constraint more visible: https://github.com/palantir/http-remoting/blob/8b1f898639ad1b7eaeb854be9dc6f84b0a44baa3/jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/feignimpl/FailoverFeignTarget.java#L133 fails when the list of servers is empty.